service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderControllerTest.java

```diff
@@ -0,0 +1,226 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C21OrderBundle;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.CreateC21OrderService;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C21;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.MAGISTRATES;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readBytes;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(C21OrderController.class)
+@OverrideAutoConfiguration(enabled = true)
+class C21OrderControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final String C21_DOCUMENT_TITLE = C21.getDocumentTitle();
+
+    @MockBean
+    private CreateC21OrderService createC21OrderService;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void aboutToStartShouldReturnErrorsWhenFamilymanNumberIsNotProvided() throws Exception {
```

Comment:
Convention is our tests begin with 'Should' and 'When' to describe the condition. I.e ShouldReturnAnErrorWhenFamilyManIsNotProvided

Reply:
Instead of splitting the about-to-start/mid-event/about-to-submit tests into 3 separate files, I've combined them all into one file, and then at the start of each test name I've stated which endpoint this test is for - see DraftOrdersControllerTest for inspiration

Reply:
Ah ok! Makes sense!